### ISSUE TITLE
Issue for the case test_2d_square_droplet in the current master version

### ISSUE THREAD

**Issue by Shuaihao-Zhang** [Xiangyu-Hu/SPHinXsys#288]
In the current master version, the calculation result for the test_2d_square_droplet case is incorrect.
![image](https://github.com/Xiangyu-Hu/SPHinXsys/assets/43446342/cdbb9c4b-9727-43d0-afaf-7cf7390074d9)

I checked the pull request history to find the potential reason.
![image](https://github.com/Xiangyu-Hu/SPHinXsys/assets/43446342/1ff98b47-bd96-441e-819a-5e9987d4a3eb)

It was found that the result is still ok after #269 is merged. The source code of #271 can not be founded, but it just changed a few lines of code. After #274 is merged. The result become unreasonable.

I haven't found the exact cause of this issue yet. I will try to figure it out. And add a regression test for this case immediately after it is fixed.

**Comment by Xiangyu-Hu** [comment]
Probably due to the change of free surface density re initialization, 
                                return SMAX(rho_sum, rho_0);

**Comment by Xiangyu-Hu** [comment]
However, may be it is not directly due to  this change.

**Comment by RussellSyne** [comment]
Thanks. I will check that first.

**Comment by RussellSyne** [comment]
> Probably due to the change of free surface density re initialization, return SMAX(rho_sum, rho_0);

Yes, this is the reason for the issue. Should I change this function `ReinitializedDensity` back to the previous version? I can submit a pull request together with a regression test for this case.

**Comment by Xiangyu-Hu** [comment]
No. That will lead to another issue of lifting of free surface in free surface flows. We may only change it for two phase flow problems.

**Comment by Xiangyu-Hu** [comment]
I have created a branch for Issue  "288-issue-for-the-case-test_2d_square_droplet in the current master version", so that we can work on it to solve the problem.

**Comment by Xiangyu-Hu** [comment]
[2021-water-13-02684.pdf](https://github.com/Xiangyu-Hu/SPHinXsys/files/11495478/2021-water-13-02684.pdf)
we may use the idea of this paper to obtained multiphase density.

**Comment by Xiangyu-Hu** [comment]
The previous formulation is OK, but seems not generalized enough (probably only works for water-air case).

**Comment by RussellSyne** [comment]
I see. Thanks!

**Comment by Xiangyu-Hu** [comment]
the new version has no such issue.
